benchmarking/generate_l1_svm_lp.jl

```diff
@@ -0,0 +1,180 @@
+  min  sum w[i] + regularizer_weight * sum z[i]
+  s.t. w >= 1 - diag(y) * [X 1] * beta
+       w >= 0
+       z >= beta
+       z >= -beta
+"""
+function populate_libsvm_model(
+  model::JuMP.Model,
+  data::SvmTrainingData,
+  regularizer_weight::Float64,
+)
+  n, d = size(data.feature_matrix)
+  println("Generating a model with $n datapoints and $(d - 1) features.")
+  JuMP.@variable(model, beta[i = 1:d])
+  JuMP.@variable(model, w[i = 1:n], lower_bound = 0.0)
+  JuMP.@variable(model, z[i = 1:d])
+  JuMP.@objective(model, Min, sum(w) + regularizer_weight * sum(z))
+  JuMP.@constraint(model, z .>= beta)
+  JuMP.@constraint(model, z .>= -beta)
+  JuMP.@constraint(
+    model,
+    w .>= (1.0 .- data.labels .* (data.feature_matrix * beta))
+  )
+  return model
+end
+
+"""
+Defines parses and args.
+
+# Returns
+A dictionary with the values of the command-line arguments.
+"""
+function parse_command_line()
+  arg_parse = ArgParse.ArgParseSettings()
+
+  ArgParse.@add_arg_table! arg_parse begin
+    "--input_filename"
+    help = "The .train file containing the problem data."
+    arg_type = String
+    required = true
+
+    "--output_filename"
+    help = "Filename for the output .mps (or .mps.gz) model."
+    arg_type = String
+    required = true
+
+    "--regularizer_weight"
+    help = "Weight of the L1 regularizer."
+    arg_type = Float64
+    required = true
+  end
+
+  return ArgParse.parse_args(arg_parse)
+end
+
+"""
+Loads a LIBSVM file and returns the loaded SvmTrainingData struct.
+"""
+function load_libsvm_file(file_name::String)
+  open(file_name, "r") do io
+    labels = Vector{Float64}()
+    row_indices = Vector{Int64}()
+    col_indices = Vector{Int64}()
+    matrix_values = Vector{Float64}()
+
+    row_index = 0
+    for line in eachline(io)
+      row_index += 1
+      split_line = split(line)
+
+      label = parse(Float64, split_line[1])
+      # This ensures that labels are 1 or -1. Different datasets use {-1, 1}, {0, 1}, and {1, 2}.
+      if label ≈ 1.0
+        label = 1.0
+      else
+        label = -1.0
+      end
+      push!(labels, label)
+
+      for i in 2:length(split_line)
+        push!(row_indices, row_index)
+        matrix_coef = split(split_line[i], ":")
+        push!(col_indices, parse(Int64, matrix_coef[1]))
+        push!(matrix_values, parse(Float64, matrix_coef[2]))
+      end
+    end
+    feature_matrix = sparse(row_indices, col_indices, matrix_values)
+    return SvmTrainingData(feature_matrix, labels)
+  end
+end
+
+function normalize_columns(feature_matrix::SparseMatrixCSC{Float64,Int64})
+  norm_of_columns = vec(sqrt.(sum(t -> t^2, feature_matrix, dims = 1)))
+  norm_of_columns[iszero.(norm_of_columns)] .= 1.0
+  return feature_matrix * Diagonal(1.0 ./ norm_of_columns)
+end
+
+function remove_empty_columns(feature_matrix::SparseMatrixCSC{Float64,Int64})
+  keep_cols = Vector{Int64}()
+  for j in 1:size(feature_matrix, 2)
+    if length(feature_matrix[:, j].nzind) > 0
```

Comment:
This constructs a new sparse matrix just to check if the column is empty. Use `length(nzrange(feature_matrix, j))` instead.